### DISCUSSION TITLE
How to checkout a certain revision

### DISCUSSION THREAD

**Discussion by roele** [GitoxideLabs/gitoxide#1446] [Q&A] [answered]
I am currently investigating to replace git2 with gitoxide and struggling to find a replacement for following code.

```
let repo = git2::clone(package_repo.url.as_str(), &tmp_repo_dir)?;
let (object, reference) = repo.revparse_ext(revision)?;
repo.checkout_tree(&object, None)?;
repo.set_head(reference.unwrap().name().unwrap())?;
```

From what i've seen in other discussions and google search results it's not clear to me if this is currently supported.

**Comment by Byron** [comment, marked as answer]
Unfortunately a checkout isn't implemented yet. Please take a look at the [Feature Discovery](https://github.com/Byron/gitoxide?tab=readme-ov-file#feature-discovery) section of the README for an idea about the macro-features that are already available.

My recommendation is to wait until `gitoxide` is stable as `git2` will continue to be maintained, unless there is a very good reason for risking an increased maintenance burden due to occasional breaking changes.

**Reply by roele** [reply]
Thanks for the quick reply.
